### PR TITLE
Add Codecov support for the Android app.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Fixes #

**Description of the Change**
Add support for generating code coverage reports for use with codecov.io in the Android app.

**Release Notes**
N/A

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ addons:
 
 env:
   global:
+    - ANDROID_TARGET=android-25
```
Comment (AenBleidd) [inline .travis.yml]: These two lines are from different PR and should be removed

File: .travis.yml [hunk 2]
```diff
@@ -58,6 +60,7 @@ jobs:
             - android-29
             - extra-google-m2repository
             - extra-android-m2repository
+            - sys-img-${ANDROID_ABI}-${ANDROID_TARGET}
```
Comment (AenBleidd) [inline .travis.yml]: This is a line from another PR too.

File: .travis.yml [hunk 3]
```diff
@@ -69,6 +72,7 @@ before_install:
 before_script:
 - if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then ( ./_autosetup ) fi
 - if [[ "${BOINC_TYPE}" == "integration-test" ]]; then ( ./integration_test/installTestSuite.sh ) fi
+- if [[ "${BOINC_TYPE}" == "manager-android" ]]; then ( echo no | avdmanager create avd --force -n test -k "system-images;ANDROID_TARGET;$ANDROID_ABI" && emulator -avd test -no-skin -no-audio -no-window &; ) fi
```
Comment (AenBleidd) [inline .travis.yml]: We don't need to run emulator in this PR

File: .travis.yml [hunk 4]
```diff
@@ -80,9 +84,10 @@ script:
 - if [[ "${BOINC_TYPE}" == "libs-mingw" ]]; then ( cd lib && MINGW=x86_64-w64-mingw32 make -f Makefile.mingw ) fi
 - if [[ "${BOINC_TYPE}" == "apps-mingw" ]]; then ( cd lib && MINGW=x86_64-w64-mingw32 make -f Makefile.mingw wrapper ) fi
 - if [[ "${BOINC_TYPE}" == "manager-osx" ]]; then ( ./3rdParty/buildMacDependencies.sh -q && ./mac_build/buildMacBOINC-CI.sh --no_shared_headers ) fi
-- if [[ "${BOINC_TYPE}" == "manager-android" ]]; then ( cd android && ./buildAndroidBOINC-CI.sh --arch arm --silent && ./buildAndroidBOINC-CI.sh --arch arm64 --silent && ./buildAndroidBOINC-CI.sh --arch x86 --silent && ./buildAndroidBOINC-CI.sh --arch x86_64 --silent && cd BOINC && ./gradlew clean build && cd ../../ ) fi
+- if [[ "${BOINC_TYPE}" == "manager-android" ]]; then ( cd android && ./buildAndroidBOINC-CI.sh --arch arm --silent && ./buildAndroidBOINC-CI.sh --arch arm64 --silent && ./buildAndroidBOINC-CI.sh --arch x86 --silent && ./buildAndroidBOINC-CI.sh --arch x86_64 --silent && cd BOINC && ./gradlew clean build jacocoTestReport assembleAndroidTest && android-wait-for-emulator && adb shell setprop dalvik.vm.dexopt-flags v=n,o=v && ./gradlew connectedCheck && cd ../../ ) fi
```
Comment (AenBleidd) [inline .travis.yml]: There should be no code related to emulator run and corresponding tests

File: .travis.yml [hunk 5]
```diff
@@ -80,9 +84,10 @@ script:
...
 - if [[ "${BOINC_TYPE}" == "unit-test" ]]; then ( ./3rdParty/buildLinuxDependencies.sh --gtest-only && ./configure --disable-client --disable-manager --enable-unit-tests CFLAGS="-g -O0" CXXFLAGS="-g -O0" && make && ./tests/executeUnitTests.sh --report-coverage ) fi
 - if [[ "${BOINC_TYPE}" == "integration-test" ]]; then ( ./integration_test/executeTestSuite.sh ) fi
 
 after_success:
 - ./deploy/prepare_deployment.sh ${BOINC_TYPE} deploy/${BOINC_TYPE}/ && ./deploy/deploy_to_bintray.sh deploy/${BOINC_TYPE}/
+- if [[ "${BOINC_TYPE}" == "manager-android" ]]; then ( cd android/BOINC && bash <(curl -s https://codecov.io/bash) ) fi
```
Comment (AenBleidd) [inline .travis.yml]: Are you sure it's done this way? I read in the article I gave to you that it is a method for CI that a not support codecov by default. Travis CI do this, so, I'm afraid, another colde should me add here

Comment (Isira-Seneviratne) [reply .travis.yml]: I got that line from this:

https://github.com/codecov/example-android/blob/master/.travis.yml

Comment (AenBleidd) [reply .travis.yml]: Could you please take a look how it's done here: https://github.com/BOINC/boinc/blob/c3fcf3322a57d80e8a6a9404fbeed7afd1affe15/.travis.yml#L84 ?

Comment (Isira-Seneviratne) [reply .travis.yml]: Ah, my apologies.

**Pull request merged**